chain/src/main/scala/org/bitcoins/chain/validation/TipValidation.scala

```diff
@@ -0,0 +1,47 @@
+package org.bitcoins.chain.validation
+
+import org.bitcoins.chain.blockchain.Blockchain
+import org.bitcoins.chain.models.{BlockHeaderDb, BlockHeaderDbHelper}
+import org.bitcoins.core.protocol.blockchain.BlockHeader
+import org.bitcoins.core.util.BitcoinSLogger
+
+sealed abstract class TipValidation extends BitcoinSLogger {
+
+  /** Checks if the given header can be connected to the current tip
+    * This is the method where a [[BlockHeader]] is transformed into a
+    * [[BlockHeaderDb]]. What this really means is that a height is
+    * assigned to a [[BlockHeader header]] after all these
+    * validation checks occur
+    * */
+  def checkNewTip(
+      newPotentialTip: BlockHeader,
+      currentTip: BlockHeaderDb): TipUpdateResult = {
+    val header = newPotentialTip
+    logger.debug(
+      s"Checking header=${header.hashBE.hex} to try to connect to currentTip=${currentTip.hashBE.hex}")
+    if (header.previousBlockHashBE != currentTip.hashBE) {
+      logger.warn(
+        s"Failed to connect tip=${header.hashBE.hex} to current chain")
+      TipUpdateResult.BadPreviousBlockHash(newPotentialTip)
+    } else if (header.nBits != currentTip.nBits) {
+      //TODO: THis is a bug, this should only occurr only 2016 blocks
+      //also this doesn't apply on testnet/regtest
+      TipUpdateResult.BadPOW(newPotentialTip)
```

Comment:
Is the intention of `BadPOW` to also check if sufficient work is performed? Or is that done elsewhere? 

Anyway I think it would be an idea to do the same you did with `isBadNonce` and move it into a method that can be clearly marked as WIP